modules/aleph/src/Aleph/Handler/AlephPatronHandler.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Drupal\aleph\Aleph\Handler;
+
+use Drupal\aleph\Aleph\AlephMaterial;
+use Drupal\aleph\Aleph\AlephPatron;
+use Drupal\aleph\Aleph\AlephClient;
+use Drupal\aleph\Aleph\AuthenticationResult;
+
+/**
+ * Class AlephPatronHandler.
+ *
+ * @property bool authenticated
+ * @package Drupal\aleph\Aleph\Handler
+ */
+class AlephPatronHandler extends AlephHandlerBase {
+
+  protected $client;
+  private $patron;
+
+  /**
+   * AlephPatronHandler constructor.
+   *
+   * @param \Drupal\aleph\Aleph\AlephClient $client
+   *    The Aleph client.
+   */
+  public function __construct(AlephClient $client) {
+    parent::__construct($client);
+    $this->client = $client;
+  }
+
+  /**
+   * Authenticate user from Aleph.
+   *
+   * @param string $bor_id
+   *    The user ID (z303-id).
+   * @param string $verification
+   *    The user pin-code/verification code.
+   *
+   * @return \Drupal\aleph\Aleph\AuthenticationResult
+   *    The authenticated Aleph patron.
+   */
+  public function authenticate($bor_id, $verification) {
+    $response = $this->client->authenticate($bor_id, $verification);
+    $result = new AuthenticationResult($this->client, $bor_id, $verification);
+    if ($result->isAuthenticated()) {
+      $patron = new AlephPatron();
+      $patron->setId($bor_id);
+      $patron->setVerification($verification);
+      $patron->setName((string) $response->xpath('z303/z303-name')[0]);
```

Comment:
if you end up needing more data from the xml-response, consider using a factory.